vqpy/impl/vobj_base.py

```diff
@@ -5,20 +5,21 @@
 from ..base.interface import VObjBaseInterface
 from ..function import infer
 from ..impl.frame import Frame
+from ..utils.video import FrameStream
 
 
 class VObjBase(VObjBaseInterface):
     """The VObject Base Class.
     The tracker is responsible to keep objects updated when the track is active
     """
 
-    def __init__(self, frame: Frame):
-        self._frame = frame
-        self._ctx = frame.ctx
-        self._start_idx = frame.ctx.frame_id
+    def __init__(self, ctx: FrameStream):
+        self._ctx = ctx
+        self._start_idx = ctx.frame_id
         self._track_length = 0
         self._datas: List[Optional[Dict]] = []
         self._registered_names: List[str] = []
+        self._registered_names_type: Dict[str, str] = {}
```

Comment:
Shall we use two variables for `property` and `cross_vobj_property`?

Reply:
Separated.
`@property` is registered in `_registered_names`, `@cross_vobj_property` is registered in `__registered_cross_vobj_names`.